pkg/core/storage/boltdb_store.go

```diff
@@ -0,0 +1,136 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/etcd-io/bbolt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+// BoltDBOptions configuration for boltdb.
+type BoltDBOptions struct {
+	FilePath string `yaml:"FilePath"`
+}
+
+// Bucket represents bucket used in boltdb to store all the data.
+var Bucket = []byte("DB")
+
+// BoltDBStore it is the storage implementation for storing and retrieving
+// blockchain data.
+type BoltDBStore struct {
+	db *bbolt.DB
+}
+
+// BoltDBBatch simple batch implementation to satisfy the Store interface.
+type BoltDBBatch struct {
+	mem map[*[]byte][]byte
+}
+
+// Len implements the Batch interface.
+func (b *BoltDBBatch) Len() int {
+	return len(b.mem)
+}
+
+// Put implements the Batch interface.
+func (b *BoltDBBatch) Put(k, v []byte) {
+	b.mem[&k] = v
+}
+
+// NewBoltDBStore returns a new ready to use BoltDB storage with created bucket.
+func NewBoltDBStore(ctx context.Context, cfg BoltDBOptions) (*BoltDBStore, error) {
+	var opts *bbolt.Options       // should be exposed via BoltDBOptions if anything needed
+	fileMode := os.FileMode(0600) // should be exposed via BoltDBOptions if anything needed
+	fileName := cfg.FilePath
+	dir := path.Dir(fileName)
+	err := os.MkdirAll(dir, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf("could not create dir for BoltDB: %v", err)
+	}
+	db, err := bbolt.Open(fileName, fileMode, opts)
```

Comment:
True, it will fail in case of missing dir.
I checked such cases(in tests):
FilePath: "./" (will create default file)
FilePath: "./dir" (will create default file)
FilePath: "./dir/filename" (happy path)
FilePath: "./dir/filename/" (mistake in having '/') will lead to error as expected